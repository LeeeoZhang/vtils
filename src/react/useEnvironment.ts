import { getEnvironment, GetEnvironmentResult } from '../utils'
import { useMemo } from 'react'

/**
 * 获取运行环境信息。
 *
 * @returns 返回运行环境信息
 */
export function useEnvironment(): GetEnvironmentResult {
  return useMemo(getEnvironment, [])
}
