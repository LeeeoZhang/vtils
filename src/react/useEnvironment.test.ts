import { renderHook } from '@testing-library/react-hooks'
import { useEnvironment } from './useEnvironment'

describe('useEnvironment', () => {
  test('表现正常', () => {
    const { result } = renderHook(() => useEnvironment())
    expect(result.current).toMatchSnapshot()
  })
})
