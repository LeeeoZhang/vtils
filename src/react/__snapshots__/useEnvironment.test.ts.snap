// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useEnvironment 表现正常 1`] = `
Object {
  "android": false,
  "browser": true,
  "deno": false,
  "ios": false,
  "miniProgram": false,
  "nodejs": true,
  "taro": false,
  "wechatMiniProgram": false,
  "wechatWebView": false,
}
`;
